### ISSUE TITLE
dotnet 7, winforms, _get_columnCount is not implemented

### ISSUE THREAD

**Issue by beqabeqa473** [nvaccess/nvda#15283]
### Steps to reproduce:
* create a new winforms project;
* add a listview to a form;
*Add multiple columns to a listview;
* Create a new item and add subItems.

### Actual behavior:

NVDA cannot get information about existing columns

### Expected behavior:

NVDA should be able to fetch information about columns. Narrator and jaws can do it.
### NVDA logs, crash dumps and other attachments:
Traceback (most recent call last):
  File "scriptHandler.pyc", line 295, in executeScript
  File "globalCommands.pyc", line 1415, in script_navigatorObject_firstChild
  File "baseObject.pyc", line 62, in __get__
  File "baseObject.pyc", line 168, in _getPropertyViaCache
  File "NVDAObjects\behaviors.pyc", line 825, in _get_firstChild
  File "NVDAObjects\behaviors.pyc", line 820, in _makeCell
  File "baseObject.pyc", line 62, in __get__
  File "baseObject.pyc", line 168, in _getPropertyViaCache
  File "NVDAObjects\behaviors.pyc", line 791, in _get_childCount
  File "baseObject.pyc", line 62, in __get__
  File "baseObject.pyc", line 168, in _getPropertyViaCache
  File "NVDAObjects\UIA\__init__.pyc", line 1917, in _get_columnCount
NotImplementedError

#### NVDA installed/portable/running from source:
Installed
#### NVDA version:
alpha-28868,a198c9b5 (2023.3.0.28868)
#### Windows version:
Windows 10 22H2 (AMD64) build 19045.3208
#### Name and version of other software in use when reproducing the issue:
An app created using windows forms.
#### Does the issue still occur after restarting your computer?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.

No
#### If NVDA add-ons are disabled, is your problem still occurring?
Yes
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Yes
## Additional information
I am attaching a sample project with prebuilt binaries
[WinFormsApp1.zip](https://github.com/nvaccess/nvda/files/12325166/WinFormsApp1.zip)

**Comment by ABuffEr** [comment]
Hi @beqabeqa473,
I tried for curiosity, as author of ColumnsReview.
Are you sure to have created the list correctly?
Because, investigating from NVDA console, I cannot see a list object, just listitems, and this sounds quite strange...

**Comment by beqabeqa473** [comment]
Yes, i created it correctly.

It seems that marking whole winforms window as isGoodUIAWindow solves a problem.

On 8/12/23, Alberto Buffolino ***@***.***> wrote:
> Hi @beqabeqa473,
> I tried for curiosity, as author of ColumnsReview.
> Are you sure to have created the list correctly?
> Because, investigating from NVDA console, I cannot see a list object, just
> listitems, and this sounds quite strange...
>
> --
> Reply to this email directly or view it on GitHub:
> https://github.com/nvaccess/nvda/issues/15283#issuecomment-1675786853
> You are receiving this because you were mentioned.
>
> Message ID: ***@***.***>


-- 
with best regards Beqa Gozalishvili
Tell: +995593454005
Email: ***@***.***
Web: https://gozaltech.org
Skype: beqabeqa473
Telegram: https://t.me/gozaltech
facebook: https://facebook.com/gozaltech
twitter: https://twitter.com/beqabeqa473
Instagram: https://instagram.com/beqa.gozalishvili

**Comment by beqabeqa473** [comment]
Further investigation showed that winforms incorrectly exposes listview for IAccessible, however i am not sure this is a bug from winforms team side.
The problem is that UIA implementation of ListItem in NVDA is incomplete and earlier it was simpler to include syslistview32 as a badUIA window and however WindowsForms.syslistview32 is beeing normalizedand passed as syslistview32 to the badUIA window comparison this is also beeing excluded.

The correct way to fix this should be to complement UIA ListItem and make it to behave like IAccessible one and remove definition from badUIA window list.
What do you think would be a correct solution?
CC @michaelDCurran  @leonardder I think you better know internals of UIA things of NVDA.

**Comment by leonardder** [comment]
I'm pretty sure the Winforms listview has a native UIA implementation nowadays. I will look into this later this week.

**Comment by beqabeqa473** [comment]
It definitely has.

I marked this window as good UIA window and it worked just great.
The task is to decide how to correctly fix this on NVDA side.
And UIA ListItem should be compemented to match behaviour of IA ListItem

On 8/14/23, Leonard de Ruijter ***@***.***> wrote:
> I'm pretty sure the Winforms listview has a native UIA implementation
> nowadays. I will look into this later this week.
>
> --
> Reply to this email directly or view it on GitHub:
> https://github.com/nvaccess/nvda/issues/15283#issuecomment-1676750629
> You are receiving this because you were mentioned.
>
> Message ID: ***@***.***>


-- 
with best regards Beqa Gozalishvili
Tell: +995593454005
Email: ***@***.***
Web: https://gozaltech.org
Skype: beqabeqa473
Telegram: https://t.me/gozaltech
facebook: https://facebook.com/gozaltech
twitter: https://twitter.com/beqabeqa473
Instagram: https://instagram.com/beqa.gozalishvili

**Comment by seanbudd** [comment]
Reopened by #15542
